human-dynamics-estimator/src/HumanDynamicsEstimator.cpp

```diff
@@ -68,53 +176,293 @@ bool HumanDynamicsEstimator::configure(yarp::os::ResourceFinder &rf)
 bool HumanDynamicsEstimator::updateModule()
-{
-    count++;
-    std::cout<<"["<<count<<"]"<< " updateModule:< HumanDynamicsEstimator >"<<std::endl;
-    return true;
+{   
+    m_count ++;
     
-    // TODO: MAP computation --> Maria's code to be integrated
-}
+    /*
+    * ------Read from ports
+    */
+    human::HumanState *tmp_state = m_humanJointConfiguration_port.read();
+            
+    if (tmp_state == NULL)
+    {
+        yError("Something wrong in reading the human configuration port!");
+        return false;
+    }
+            
+    // transform yarp Vector in iDynTree vector
+    if (!iDynTree::toiDynTree(tmp_state->positions,m_q))
+    {
+        yError("Somenthing wrong in the conversion from a YARP to iDynTree vector!");
+        return false;
+    };
 
-//---------------------------------------------------------------------
-bool HumanDynamicsEstimator::interruptModule()
-{
-    return true;
-}
+    // transform yarp Vector in iDynTree vector           
+    if (!iDynTree::toiDynTree(tmp_state->velocities,m_dq))
+    {
+         yError("Somenthing wrong in the conversion from a YARP to iDynTree vector!");
+         return false;
+    };
 
+    human::HumanForces *tmp_force = m_humanForces_port.read();
+
+    if (tmp_force == NULL)
+    {
+        yError("Something wrong in reading the human forces port!");
+        return false;
+    }
+    
+    for (std::vector<human::Force6D>::iterator it(tmp_force->forces.begin()); it!=tmp_force->forces.end(); ++it ) 
+    {
+        std::string appliedLink = it->appliedLink;
+        std::unordered_map<std::string,int>::const_iterator found = m_y_map.find(appliedLink);
+        m_y.setVal(found->second, it->fx);
+        m_y.setVal((found->second)+1, it->fy);
+        m_y.setVal((found->second)+2, it->fz);
+        m_y.setVal((found->second)+3, it->ux);
+        m_y.setVal((found->second)+4, it->uy);
+        m_y.setVal((found->second)+5, it->uz);
+    }
+    
+    //TODO: fill y with measurements from ACCELEROMETER, GYROSCOPE and DOF_ACCELERATION.
+    // At this stage they are 0!  
+
+    /*
+     * Set the kinematic information necessary for the dynamics estimation
+     */
+    m_berdy.updateKinematicsFromTraversalFixedBase(m_q,m_dq,m_gravity);
+
+    /*
+     * Get berdy matrices
+     */
+    m_berdy.getBerdyMatrices(m_D,m_bD,m_Y,m_bY);    
+    
+    /*
+     * MAP computation
+     */ 
+    //TODO: create a MAP function/class
+    Eigen::SparseMatrix<double> sigma_BarD_inv(m_dynV,m_dynV);   
+    Eigen::VectorXd rhs_BarD(m_dynV);
+    Eigen::VectorXd mu_BarD(m_dynV);    
+
+    Eigen::SparseMatrix<double> sigma_dgiveny_inv(m_dynV,m_dynV);
+    Eigen::VectorXd rhs(m_dynV);    
+    Eigen::VectorXd mu_dgiveny(m_dynV); 
+ 
+    sigma_BarD_inv = (toEigen(m_D).transpose() * toEigen(m_sigmaD_prior) * toEigen(m_D)) + toEigen(m_sigmad_prior); 
+   
+    if(m_count==1) {                    
+    m_llt.analyzePattern(sigma_BarD_inv);  
```

Comment:
Now I get why you needed `m_count`.
Use a boolean or, better, move the `analyzePatterns` to the configure